packages/mds-compliance-engine/engine/helpers.ts

```diff
@@ -60,22 +60,37 @@ export async function getAllInputs() {
  * @returns All events and devices for a given provider
  */
 export async function getProviderInputs(provider_id: string, timestamp: number = now()) {
-  const { events } = await IngestServiceClient.getEventsUsingOptions({
+  const {
+    events: eventsAcc,
+    cursor: { next }
+  } = await IngestServiceClient.getEventsUsingOptions({
     provider_ids: [provider_id],
     time_range: { start: timestamp - days(2), end: timestamp },
     grouping_type: 'latest_per_vehicle',
     order: { column: 'timestamp', direction: 'ASC' }
   })
 
-  const deviceMap = (await IngestServiceClient.getDevices(events.map(({ device_id }) => device_id))).reduce<{
+  let cursor = next
```

Comment:
Why the two variables? Couldn't we initialize it as cursor at the top of getProviderInputs and then replace it at the end of the loop as currently?